SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -796,39 +798,58 @@ void USpatialActorChannel::RemoveRepNotifiesWithUnresolvedObjs(TArray<UProperty*
 	});
 }
 
-void USpatialActorChannel::SpatialViewTick()
+void USpatialActorChannel::ServerViewTick()
 {
-	if (Actor != nullptr && !Actor->IsPendingKill() && IsReadyForReplication())
+	if (!IsAuthoritativeServer())
+	{
+		return;
+	}
+
+	UObject* ControllerCDO = AController::StaticClass()->GetDefaultObject();
+	UObject* PlayerControllerCDO = APlayerController::StaticClass()->GetDefaultObject();
+
+	FString OldOwnerWorkerAttribute = NetOwnerWorkerAttribute;
+	NetOwnerWorkerAttribute = improbable::GetOwnerWorkerAttribute(Actor);
```

Comment:
Can we not detect that an ownership change may have happened from what state we send rather than doing a string compare every tick. I don't think this will cause issues but it seems more expensive than it needs to be. Also not the place I'd expect to look for such behaviour.

Reply:
The initial implementation for this tried to utilize that, however at the time there was a problem with that approach/an edge case it couldn't handle in a nice way. Unfortunately, I can't recall the exact details right now, and I don't think there is significant benefit from changing this now.

Reply:
I'd sat the benefit is from encapsulating where we have specific SpatialOS behaviour. Right now we have it dotted all over the net driver and the receiver. Now we also have more in a specific place in the actor channel. This becomes a large problem when each piece isn't in complete isolation (and they rarely are). If this has to be ordered before something else or someone needs to upgrade ACLs to layers then this get hard to read and reason about. If we think it's too much work then this is probably fine for now, but we should schedule work to solve that.

Reply:
Will resolve the comment given that it's not asking for changes anymore.